### PR TITLE
Travis -> Github Actions

### LINKED ISSUE #938: CI fails on all Ruby versions
# Bug report

CI is red. 

**Current behavior:**

All Ruby versions in the matrix fail their build. 

**Steps to reproduce current behavior:**

See the CI. 

**Expected behavior:**

Green. 
**System information:**

* ruby version: all
* savon version: latest git HEAD

### PR DESCRIPTION
**Pull request by pcai**
Fixes #938

**What kind of change is this?**
- Migrate to Github actions for CI
- Remove coveralls ([it was breaking the CI](https://github.com/savonrb/savon/runs/1377306164?check_suite_focus=true))
- Fix a failing test
- Fix some test warnings
- Fix frozen string literal bug that a test revealed
- Add byebug in development since I use it to debug broken tests

Couple of hurdles I encountered:
- Travis config must've been broken, because there are clear test failures despite builds returning green

**Did you add tests for your changes?**

Well... :joy: 

**Summary of changes**

Reference: https://docs.travis-ci.com/user/migrate/open-source-repository-migration/#frequently-asked-questions

> Q. When will the migration from travis-ci.org to travis-ci.com be completed? #
> 
> A. In an effort to ensure that all of our users - whether you build open-source, public or private repositories - receive regular feature updates, security patches and UX/UI enhancements, we are announcing that travis-ci.org will be officially closed down completely no later than December 31st, 2020, allowing us to focus all our efforts on bringing new features and fixes to travis-ci.com and all of our awesome users like yourself on the travis-ci.com domain.
> 
> Q. What will happen to travis-ci.org after December 31st, 2020? #
> 
> A. Travis-ci.org will be switched to a read-only platform, allowing you to see your jobs build history from all repositories previously connected to travis-ci.org.
> 

**Other information**

### CODE REVIEW

File: lib/savon/builder.rb [hunk 1]
```diff
@@ -154,7 +154,7 @@ def serialized_messages
         break if @locals[:message].nil?
         message_locals = @locals[:message][message.snakecase.to_sym]
         message_content = Message.new(message_tag, namespace_identifier, @types, @used_namespaces, message_locals, :unqualified, @globals[:convert_request_keys_to], @globals[:unwrap]).to_s
-        messages << "<#{message} xsi:type=\"#{type.join(':')}\">#{message_content}</#{message}>"
+        messages += "<#{message} xsi:type=\"#{type.join(':')}\">#{message_content}</#{message}>"
```
Comment (pcai) [inline lib/savon/builder.rb:157]: under frozen string literals, this breaks

File: spec/savon/softlayer_spec.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
       locals = Savon::LocalOptions.new(message)
       builder = Savon::Builder.new(:create_object, wsdl, globals, locals)
-      expect(builder.to_s).to eq('<?xml version="1.0" encoding="UTF-8"?><env:Envelope xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:tns="http://api.service.softlayer.com/soap/v3/" xmlns:env="http://schemas.xmlsoap.org/soap/envelope/"><env:Body><tns:createObject><templateObject><longName>Zertico LLC Reseller</longName></templateObject></tns:createObject></env:Body></env:Envelope>')
+      expect(builder.to_s).to eq('<?xml version="1.0" encoding="UTF-8"?><env:Envelope xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:tns="http://api.service.softlayer.com/soap/v3/" xmlns:env="http://schemas.xmlsoap.org/soap/envelope/"><env:Body><createObject><templateObject xsi:type="tns:SoftLayer_Brand"><longName>Zertico LLC Reseller</longName></templateObject></createObject></env:Body></env:Envelope>')
```
Comment (pcai) [inline spec/savon/softlayer_spec.rb:25]: I dont know any better but since it is in master I defaulted to assuming the code is right and the test is wrong.

**Comment by pcai** [comment]
We should be able to put jruby head in the matrix after #927 too

**Review by olleolleolle — APPROVED** [review]
Thanks for polishing the engine!

**Pull request merged**